pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -964,6 +967,109 @@ type MSTeamsConfig struct {
+	// Defaults to https://open.rocket.chat/ if not specified.
+	// +optional
+	Channel *string `json:"channel,omitempty"`
+	// The sender token. Either Token or TokenFile is required.
+	// +optional
+	Token v1.SecretKeySelector `json:"token,omitempty"`
+	// The file containing the sender token. Either Token or TokenFile is required.
+	// +optional
+	TokenFile *string `json:"tokenFile,omitempty"`
+	// The sender token ID. Either TokenID or TokenIDFile is required.
+	// +optional
+	TokenID v1.SecretKeySelector `json:"tokenID,omitempty"`
+	// The file containing the sender token ID. Either TokenID or TokenIDFile is required.
+	// +optional
+	TokenIDFile *string `json:"tokenIDFile,omitempty"`
+	// The message color.
+	// Defaults to '{{ if eq .Status "firing" }}red{{ else }}green{{ end }}'.
+	// +optional
+	Color *string `json:"color,omitempty"`
+	// Emoji to use for the message.
+	// Defaults to '{{ template "rocketchat.default.emoji" . }}'.
+	// +optional
+	Emoji *string `json:"emoji,omitempty"`
+	// Icon URL for the message.
+	// Defaults to '{{ template "rocketchat.default.iconurl" . }}'.
+	// +optional
+	IconURL *string `json:"iconURL,omitempty"`
+	// The main message text.
+	// Defaults to '{{ template "rocketchat.default.text" . }}'.
+	// +optional
+	Text *string `json:"text,omitempty"`
+	// The message title.
+	// Defaults to '{{ template "rocketchat.default.title" . }}'.
+	// +optional
+	Title *string `json:"title,omitempty"`
+	// The title link for the message.
+	// Defaults to '{{ template "rocketchat.default.titlelink" . }}'.
+	// +optional
+	TitleLink *string `json:"titleLink,omitempty"`
+	// Additional fields for the message.
+	// +optional
+	Fields []RocketChatFieldConfig `json:"fields,omitempty"`
+	// Whether to use short fields.
+	// Defaults to false.
+	// +optional
+	ShortFields *bool `json:"shortFields,omitempty"`
+	// Image URL for the message.
+	// +optional
+	ImageURL *string `json:"imageURL,omitempty"`
+	// Thumbnail URL for the message.
+	// +optional
+	ThumbURL *string `json:"thumbURL,omitempty"`
+	// Whether to enable link names.
+	// +optional
+	LinkNames *bool `json:"linkNames,omitempty"`
+	// Actions to include in the message.
+	// +optional
+	Actions []RocketChatActionConfig `json:"actions,omitempty"`
+	// HTTP client configuration.
+	// +optional
+	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
+}
+
+// RocketChatFieldConfig defines additional fields for RocketChat messages.
+type RocketChatFieldConfig struct {
+	// The field title.
+	// +optional
+	Title string `json:"title,omitempty"`
+	// The field value.
+	// +optional
+	Value string `json:"value,omitempty"`
+	// Whether the field should be displayed in a compact format.
+	// +optional
+	Short *bool `json:"short,omitempty"`
+}
+
+// RocketChatActionConfig defines actions for RocketChat messages.
+type RocketChatActionConfig struct {
+	// The type of action (only "button" is supported).
+	// +optional
+	Type string `json:"type,omitempty"`
```

Comment:
```suggestion
	Type string* `json:"type,omitempty"`
```

The same for other optional fields